compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -164,6 +164,12 @@ object Applications {
 
   def wrapDefs(defs: mutable.ListBuffer[Tree], tree: Tree)(implicit ctx: Context): Tree =
     if (defs != null && defs.nonEmpty) tpd.Block(defs.toList, tree) else tree
+
+  /** A wrapper indicating that its argument is an application of an extension method.
```

Comment:
I chose a wrapper since that way we are sure we can't ignore it. It's supposed to be always eliminated. In any case Pickler would fail if it encountered one of these. That's also why I think we don't need to burden TreeChecker with checking this separately.